tests/src/test/java/org/eclipse/hono/tests/registry/CredentialsAmqpIT.java

```diff
@@ -119,37 +134,84 @@ public void testGetCredentialsNotExistingAuthId(final TestContext ctx) {
 
     /**
      * Verifies that the service returns credentials for a given type and authentication ID.
-     * 
+     *
      * @param ctx The vert.x test context.
      */
     @Test
     public void testGetCredentialsReturnsCredentialsTypeAndAuthId(final TestContext ctx) {
 
-        credentialsClient
-            .get(CredentialsConstants.SECRETS_TYPE_HASHED_PASSWORD, CREDENTIALS_AUTHID1)
-            .setHandler(ctx.asyncAssertSuccess(result -> {
-                ctx.assertEquals(CREDENTIALS_AUTHID1, result.getAuthId());
-                ctx.assertEquals(CredentialsConstants.SECRETS_TYPE_HASHED_PASSWORD, result.getType());
-            }));
+        // Prepare the credential to insert
+        final String deviceId = helper.getRandomDeviceId(Constants.DEFAULT_TENANT);
+        final JsonObject payload = new JsonObject()
+                .put(CredentialsConstants.FIELD_PAYLOAD_DEVICE_ID, deviceId)
+                .put(CredentialsConstants.FIELD_TYPE, CredentialsConstants.SECRETS_TYPE_HASHED_PASSWORD)
+                .put(CredentialsConstants.FIELD_AUTH_ID, CREDENTIALS_AUTHID1)
+                .put(CredentialsConstants.FIELD_SECRETS, getAuthId1Secrets());
+
+        final Async done = ctx.async();
+        // Insert it into the device Registry
+        helper.registry.addCredentials(Constants.DEFAULT_TENANT, payload).setHandler(r -> {
+
+            // it's successfully inserted, run the test.
```

Comment:
simply *compose()* the next step instead of using *setHandler* and checking manually if the previous step succeeded.
at the end, you can then use *setHandler(ctx.asyncAssertSuccess())* to make sure that the test succeeds. No need for the *done* `Async` as well ...